runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -185,17 +345,25 @@ public void commit() {
      * if no snapshot is specified, applies all the updates and then ends the Corfu transaction.
      * The commit returns successfully if the transaction was committed.
      * Otherwise this throws a TransactionAbortedException.
-     *
-     * @param timestamp Timestamp to commit the transaction on.
+     * @return - address at which the commit of this transaction occurred.
      */
-    public void commit(final Timestamp timestamp) {
-        this.timestamp = timestamp;
-        try {
-            txBegin();
+    public long commit() {
+        long commitAddress;
+        try { // batch up all the operations and apply them in a new transaction started here.
+            txBeginInternal();
             operations.forEach(Runnable::run);
         } finally {
-            txEnd();
-            operations.clear();
+            commitAddress = endTxn();
```

Comment:
should we set startedReadWriteTxn=false here also?

Reply:
not sure why - if readYour writes then it gets set in txnBeginInternal if not it would have started whenever the transaction would have actually begun right?